pkg/cli/cmd/bicep/publish/publish_test.go

```diff
@@ -313,3 +317,97 @@ func TestRunner_Validate(t *testing.T) {
 	}
 	radcli.SharedValidateValidation(t, NewCommand, tests)
 }
+
+type MockErrorResponse struct {
+	Method     string
+	URL        *url.URL
+	StatusCode int
+}
+
+// Error implements error.
+func (e *MockErrorResponse) Error() string {
+	// panic("unimplemented")
+	return e.Method
+}
+
+func (e *MockErrorResponse) GetMethod() string {
+	return e.Method
+}
+
+func (e *MockErrorResponse) GetURL() *url.URL {
+	return e.URL
+}
+
+func (e *MockErrorResponse) GetStatusCode() int {
+	return e.StatusCode
+}
+
+func TestHandleErrorResponse(t *testing.T) {
+	// Test case 1: Unauthorized with ACR info
```

Comment:
Use [subtests](https://go.dev/blog/subtests) for each of these. Go has a feature for this :)